kratos/python_scripts/python_solver.py

```diff
@@ -180,15 +177,64 @@ def _ImportModelPart(self, model_part, model_part_import_settings):
+
+        if input_type.IsArray():
+            if model_part_import_settings["input_filename"].IsArray():
+                current_model = model_part.GetModel()
+                combine_param = KratosMultiphysics.Parameters("""{
+                    "model_parts_list"         : [],
+                    "combined_model_part_name" : ""
+                }""")
+                combine_param["combined_model_part_name"].SetString(model_part.Name)
+                filenames_list = model_part_import_settings["input_filename"].GetStringArray()
+                copy_model_part_import_settings = KratosMultiphysics.Parameters(model_part_import_settings)
+                copy_model_part_import_settings.RemoveValue("input_filename")
+                copy_model_part_import_settings.AddString("input_filename", "")
+                for i in range(len(filenames_list)):
+                    aux_name = "AUX_MODELPART" + str(i)
+                    combine_param["model_parts_list"].Append(aux_name)
+                    aux_model_part = current_model.CreateModelPart(aux_name)
+                    copy_model_part_import_settings["input_filename"].SetString(filenames_list[i])
+                    self._single_ImportModelPart(aux_model_part, model_part_import_settings, input_type[i].GetString())
+                KratosMultiphysics.ModelPartCombinationUtilities(current_model).CombineModelParts(combine_param)
+            else:
+                raise Exception("Multiple input formats, but only one file?.")
+        elif input_type.GetString() == "mdpa": # Add other file reading formats
+            if model_part_import_settings["input_filename"].IsArray():
+                current_model = model_part.GetModel()
+                combine_param = KratosMultiphysics.Parameters("""{
+                    "model_parts_list"         : [],
+                    "combined_model_part_name" : ""
+                }""")
+                combine_param["combined_model_part_name"].SetString(model_part.Name)
+                filenames_list = model_part_import_settings["input_filename"].GetStringArray()
+                copy_model_part_import_settings = KratosMultiphysics.Parameters(model_part_import_settings)
+                copy_model_part_import_settings.RemoveValue("input_filename")
+                copy_model_part_import_settings.AddString("input_filename", "")
+                for i in range(len(filenames_list)):
+                    aux_name = "AUX_MODELPART" + str(i)
+                    combine_param["model_parts_list"].Append(aux_name)
+                    aux_model_part = current_model.CreateModelPart(aux_name)
+                    copy_model_part_import_settings["input_filename"].SetString(filenames_list[i])
+                    self._single_ImportModelPart(aux_model_part, model_part_import_settings, input_type.GetString())
+                KratosMultiphysics.ModelPartCombinationUtilities(current_model).CombineModelParts(combine_param)
```

Comment:
I really don't like this implementation as it is very messy (e.g. hardcoded names) 

Esp since the importing should be in a separate utility like it is in MPI

Furthermore since we have now a better mechanism with the Modellers I would go in this way

Suggestion:
Since I see the use of it and it is very difficult to do in MPI, why don't you add the `SerialModelPartCombinatorModeler` to make it clear that it doesn't work in MPI?
To me this would be a good intermediate solution